ac/ac-quiz/src/index.js

```diff
@@ -39,52 +40,126 @@ export const config = {
             items: {
               type: "object",
               required: [
-                "title"
+                "answer"
               ],
               properties: {
-                title: {
+                answer: {
                   type: "string",
                   title: "Answer"
-                },
-                answer: {
-                  type: "boolean",
-                  title: "This is an answer",
-                  default: false
                 }
               }
             }
-          },
-          details: {
-            type: "string",
-            title: "Enter an explanation",
           }
         }
       }
     }
   }
 }
 
-
 export const ActivityRunner = (props: ActivityRunnerT) => {
-  const { config, logger, onCompletion } = props
-
-  const propdef = config.MCQ.reduce(
-    (questionAcc, question, questionIndex) => ({...questionAcc, [questionIndex + '']: { 
-      type: 'string', 
-      title: question.title,
-      description: question.details,
-      enum: question.answers.reduce((answerAcc, answer, answerIndex) => ([...answerAcc, answer.title]), {})
-    }}),
-    {} )
-  const formdef = { 
+  const { config, saveProduct, object, userInfo, reactiveData, reactiveFn } = props
+  const { products, socialStructures } = object
+
+  const schema = {
     title: config.name,
     type: 'object',
-    properties: propdef
+    properties: {}
   }
-  const uiSchema = { 'MCQ': { 'ui:options': { 'backgroundColor': 'pink' } } }
+  
+  const uiSchema = {
+    'MCQ': { 'ui:options': { 'backgroundColor': 'pink' } }
+  }
+  
+  config.MCQ.forEach(
+    (question, questionIndex) => {
+      schema.properties[''+questionIndex] = { 
+        type: 'object',
+        title: 'Question ' + (1 + questionIndex),
+        required: [
+          'justification'
+        ],
+        properties: {
+          'radio': {
+            type: 'string', 
+            title: question.question,
+            enum: question.answers.map((answer, answerIndex) => (answer.answer))
+          },
+          'justification': {
+            type: 'string',
+            title: 'Explain your answer'
+          }
+        }
+      }
+      uiSchema[''+questionIndex] = { 
+        radio: { "ui:widget": "radio" } 
+      }
+    }
+  )
+
+  const socialStructure = socialStructures.find(x => x[userInfo.id]) 
+
+  const groupId = socialStructure 
+    ? socialStructure[userInfo.id].group
+    : 'NO_GROUP'
+
+  const reactiveKey = reactiveData.keys.find(x => x.groupId === groupId)
+  const formData = reactiveKey
+    ? reactiveKey['DATA' + userInfo.id]
+    : null
+
+  const findPartner = socialStructure
+    ? Object.keys(socialStructure).find(
+      id => !(id === userInfo.id) && socialStructure[id].group === groupId
+    )
+    : null
+
+  const partnerId = findPartner ? findPartner : 'NO_ID'
+
+  const partnerFormData = reactiveKey
```

Comment:
I'm surprised to see that flow does not complain on that. I remember it did not like that I multiply Boolean with Integers :smile: